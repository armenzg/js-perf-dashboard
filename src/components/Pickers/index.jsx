import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Picker from '../Picker';
import { BENCHMARKS, CONFIG } from '../../config';
import { generateLastDaysLabel } from '../../utils/timeRangeUtils';

const styles = () => ({
  root: {
    display: 'flex',
    textAlign: 'center',
    padding: '15px',
  },
});

const Pickers = ({
  classes, benchmark, category, onChange, platform, dayRange, resultsBenchmarks,
}) => (
  <div className={classes.root}>
    <Picker
      key="Platform selection"
      identifier="platform"
      topLabel="Platform"
      onSelection={onChange}
      selectedValue={platform}
      options={
        Object.keys(CONFIG.views).reduce((res, platformKey) => {
          res.push({ value: platformKey, label: CONFIG.views[platformKey].label });
          return res;
        }, [])
      }
    />
    <Picker
      key="Category selection"
      identifier="category"
      topLabel="Category"
      onSelection={onChange}
      selectedValue={category}
      options={
            Object.entries(CONFIG.views[platform].categories).map(
              ([catKey, catValue]) => ({ value: catKey, label: catValue.label }),
            )
        }
    />
    {CONFIG.views[platform].categories[category] && (
    <Picker
      key="Benchmark selection"
      identifier="results"
      topLabel="Results"
      onSelection={onChange}
      selectedValue={benchmark}
      options={
        resultsBenchmarks.sort().reduce((res, benchmarkKey) => {
          res.push({
            value: benchmarkKey,
            label: BENCHMARKS[benchmarkKey].label,
          });
          return res;
        }, [{ value: 'overview', label: 'Overview' }])
      }
    />
    ) }
    <Picker
      key="Time range"
      identifier="numDays"
      topLabel="Time range"
      onSelection={onChange}
      selectedValue={dayRange}
      options={CONFIG.dayRange.map((numDays) => ({
        value: numDays,
        label: generateLastDaysLabel(numDays),
      }))}
    />
  </div>
);

Pickers.propTypes = ({
  classes: PropTypes.shape().isRequired,
  benchmark: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  platform: PropTypes.string.isRequired,
  dayRange: PropTypes.number.isRequired,
});

export default withStyles(styles)(Pickers);
