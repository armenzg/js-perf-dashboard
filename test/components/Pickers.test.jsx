import React from 'react';
import { render } from '@testing-library/react';
import 'raf/polyfill';
import Pickers from '../../src/components/Pickers';

it('renders correctly', () => {
  const { asFragment } = render(
    <Pickers platform="win10" benchmark="overview" onChange={() => {}} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
