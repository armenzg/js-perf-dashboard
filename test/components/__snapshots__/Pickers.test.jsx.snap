// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Pickers-root-1"
  >
    <form
      autocomplete="off"
      class="Picker-root-2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Platform
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
            id="select-platform"
            role="button"
            tabindex="0"
          >
            Windows 10 64bit
          </div>
          <input
            name="platform"
            type="hidden"
            value="win10"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </form>
    <form
      autocomplete="off"
      class="Picker-root-2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Benchmark
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
            id="select-benchmark"
            role="button"
            tabindex="0"
          >
            Overview
          </div>
          <input
            name="benchmark"
            type="hidden"
            value="overview"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
